Mathlib/CategoryTheory/Monoidal/Mod_.lean

```diff
@@ -129,3 +128,35 @@ def comap {A B : Mon_ C} (f : A ⟶ B) : Mod_ B ⥤ Mod_ A where
 -- Lots more could be said about `comap`, e.g. how it interacts with
 -- identities, compositions, and equalities of monoid object morphisms.
 end Mod_
+
+section Mod_Class
+
+open CategoryTheory Mon_Class MonoidalCategory
+
+variable (M : C) [Mon_Class M]
+
+/-- An action of a monoid object `M` on an object `X` is the data of a map `smul : M ⊗ X ⟶ X` that
+satisfies unitality and associativity with multiplication.
+
+See `MulAction` for the non-categorical version. -/
+class Mod_Class (X : C) where
+  /-- The action map -/
+  smul : M ⊗ X ⟶ X
+  /-- The identity acts trivially. -/
+  one_smul (X) : (η ▷ X) ≫ smul = (λ_ X).hom := by aesop_cat
+  /-- The action map is compatible with multiplication. -/
+  mul_smul (X) : (μ ▷ X) ≫ smul = (α_ M M X).hom ≫ (M ◁ smul) ≫ smul := by aesop_cat
+
+attribute [reassoc (attr := simp)] Mod_Class.mul_smul Mod_Class.one_smul
+
```

Comment:
Can you add an instance for `Mod_Class X.X` for `X : Mod_ A`, and a constructor taking `Mod_Class` and spitting out `Mod_`? Thanks.